File: helixtrajectoryj/src/main/java/org/team2363/helixtrajectory/Obstacle.java
package org.team2363.helixtrajectory;

import static org.team2363.util.ObjectChecker.requireNonNullAndWrapUnmodifiable;

import java.util.List;

public final class Obstacle {

    public final double safetyDistance;
    public final boolean applyToAllSegments;
    public final List<? extends ObstaclePoint> points;

    public Obstacle(double safetyDistance, boolean applyToAllSegments,
            List<? extends ObstaclePoint> points) throws NullPointerException {
        this.safetyDistance = safetyDistance;
        this.applyToAllSegments = applyToAllSegments;
        this.points = requireNonNullAndWrapUnmodifiable(points, "List of obstacle points cannot be null", "Obstacle point cannot be null");
    }

    @Override
    public String toString() {
        return "{\"safety_distance\": " + safetyDistance
            + ", \"apply_to_all_segments\": " + applyToAllSegments
            + ", \"points\": " + points
            + "}";
    }
}

File: helixtrajectorycpp/src/cpp/obstacle/Obstacle.cpp
#include "obstacle/Obstacle.h"

#include <iostream>
#include <vector>

#include "obstacle/ObstaclePoint.h"

namespace helixtrajectory {

    Obstacle::Obstacle(double safetyDistance, const std::vector<ObstaclePoint>& points)
        : safetyDistance(safetyDistance), points(points) {
    }

    std::ostream& operator<<(std::ostream& stream, const Obstacle& obstacle) {
        stream << std::boolalpha;
        return stream << "{\"safety_distance\": " << obstacle.safetyDistance
                // << ", \"points\": " << obstacle.points
                << "}";
    }
}